### PR TITLE
BCI Java test for System module

### PR DESCRIPTION
**Pull request by ilausuch**
https://progress.opensuse.org/issues/110689
VR: http://marvin5.arch.suse.cz/tests/819

### CODE REVIEW

File: tests/test_openjdk.py [hunk 1]
```diff
@@ -67,28 +67,38 @@ def test_jdk_version(container, java_version):
     indirect=["container_per_test"],
 )
 @pytest.mark.parametrize(
-    "test_to_run, expected_strings, expected_exit_status, java_params",
+    "test_to_run, expected_strings, expected_exit_status, java_params, arguments, environment",
```
Comment (grisu48) [inline tests/test_openjdk.py:70]: I don't know if it is possible, but I would separate the arguments, environments tests into different functions. Here we have one mighty "it does everything" function, which decreases the visibility on the test failures.

Comment (grisu48) [reply tests/test_openjdk.py:70]: They are appearing as separate test runs, see the `openjdk` test runs:

```
tests/test_openjdk.py::test_jdk_extended[local-threads_concurrency_and_sleep-expected_strings0-expected_exit_status0----bci/openjdk:11 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:11] PASSED [ 15%]
tests/test_openjdk.py::test_jdk_extended[local-threads_concurrency_and_sleep-expected_strings0-expected_exit_status0----bci/openjdk:17 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:17] PASSED [ 20%]
tests/test_openjdk.py::test_jdk_extended[local-time-expected_strings1-expected_exit_status1----bci/openjdk:11 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:11] PASSED [ 25%]
tests/test_openjdk.py::test_jdk_extended[local-time-expected_strings1-expected_exit_status1----bci/openjdk:17 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:17] PASSED [ 30%]
tests/test_openjdk.py::test_jdk_extended[local-files_and_directories-expected_strings2-expected_exit_status2----bci/openjdk:11 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:11] PASSED [ 35%]
tests/test_openjdk.py::test_jdk_extended[local-files_and_directories-expected_strings2-expected_exit_status2----bci/openjdk:17 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:17] PASSED [ 40%]
tests/test_openjdk.py::test_jdk_extended[local-memory-expected_strings3-expected_exit_status3--Xmx10M---bci/openjdk:11 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:11] PASSED [ 45%]
tests/test_openjdk.py::test_jdk_extended[local-memory-expected_strings3-expected_exit_status3--Xmx10M---bci/openjdk:17 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:17] PASSED [ 50%]
tests/test_openjdk.py::test_jdk_extended[local-garbage_collector-expected_strings4-expected_exit_status4----bci/openjdk:11 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:11] PASSED [ 55%]
tests/test_openjdk.py::test_jdk_extended[local-garbage_collector-expected_strings4-expected_exit_status4----bci/openjdk:17 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:17] PASSED [ 60%]
tests/test_openjdk.py::test_jdk_extended[local-system_exit-expected_strings5-expected_exit_status5--2--bci/openjdk:11 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:11] PASSED [ 65%]
tests/test_openjdk.py::test_jdk_extended[local-system_exit-expected_strings5-expected_exit_status5--2--bci/openjdk:17 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:17] PASSED [ 70%]
tests/test_openjdk.py::test_jdk_extended[local-system_out-expected_strings6-expected_exit_status6----bci/openjdk:11 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:11] PASSED [ 75%]
tests/test_openjdk.py::test_jdk_extended[local-system_out-expected_strings6-expected_exit_status6----bci/openjdk:17 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:17] PASSED [ 80%]
tests/test_openjdk.py::test_jdk_extended[local-system_properties-expected_strings7-expected_exit_status7--Dproperty1=test---bci/openjdk:11 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:11] PASSED [ 85%]
tests/test_openjdk.py::test_jdk_extended[local-system_properties-expected_strings7-expected_exit_status7--Dproperty1=test---bci/openjdk:17 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:17] PASSED [ 90%]
tests/test_openjdk.py::test_jdk_extended[local-system_env-expected_strings8-expected_exit_status8---ENV1=test-bci/openjdk:11 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:11] PASSED [ 95%]
tests/test_openjdk.py::test_jdk_extended[local-system_env-expected_strings8-expected_exit_status8---ENV1=test-bci/openjdk:17 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:17] PASSED [100%]
```

So issue is resolved, this is already the case

**Review by grisu48 — APPROVED** [review]

File: tests/test_openjdk.py (hunk 1 above)
Comment (ilausuch) [reply tests/test_openjdk.py]: I follow @dcermak recommendation here.

File: tests/test_openjdk.py [hunk 2]
```diff
@@ -107,18 +119,22 @@ def test_jdk_extended(
...
+        environment
+        + " java "
         + java_params
         + " "
         + container_test_dir
         + test_to_run
-        + ".java 2>&1",
+        + ".java "
+        + arguments
+        + " 2>&1",
```
[the remark this answers is not in the document]
Comment (ilausuch) [reply tests/test_openjdk.py]: done

**Comment by ilausuch** [comment]
Also I removed System.out System.err tests because these are already included on the other tests

File: tests/test_openjdk.py [hunk 3]
```diff
@@ -61,42 +63,57 @@ def test_jdk_version(container, java_version):
     )
 
 
+@dataclass
+class TestExtendedParams:
```
Comment (dcermak) [inline tests/test_openjdk.py:67]: Could you please document the class parameters? At the moment one has to look into the code to find out what format is expected with e.g. `environment`

**Review by dcermak — COMMENTED** [review]
Looks pretty good. I've added some comments to the `TestExtendedParams` class.

File: tests/test_openjdk.py [hunk 4]
```diff
@@ -61,42 +63,57 @@ def test_jdk_version(container, java_version):
...
+    expected_strings: list = field(default_factory=lambda: [])
+    expected_err_strings: list = field(default_factory=lambda: [])
+    expected_exit_status: list = field(default_factory=lambda: [0])
```
[the remark this answers is not in the document]
Comment (ilausuch) [reply tests/test_openjdk.py]: fixed

File: tests/test_openjdk.py [hunk 5]
```diff
@@ -61,42 +63,69 @@ def test_jdk_version(container, java_version):
...
+    """
+    A class used to pass parameters and options to test_jdk_extended
+
+    Attributes:
+        expected_strings Expected standard output strings
+        expected_err_strings Expected standard error strings
+        expected_exit_status Expected list of exit codes
+        java_params java runtime parameters
+        arguments arguments to pass to the java test
+        environment environment variables set before execute the test
```
Comment (dcermak) [inline tests/test_openjdk.py:77]:
```suggestion
        expected_exit_status Expected list of allowed exit codes
        java_params java runtime parameters as they would be passed on the CLI
        arguments arguments to pass to the java test
        environment environment variables in the form ``VAR=foo VAR2=bar``
```

**Review by dcermak — COMMENTED** [review]
LGTM 👍, please just rebase on main.

**Comment by ilausuch** [comment]
Done. Thanks Dan!

File: tests/test_openjdk.py (hunk 5 above)
Comment (ilausuch) [reply tests/test_openjdk.py]: done

**Review by jlausuch — DISMISSED** [review]

File: tests/test_openjdk.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (ilausuch) [reply tests/test_openjdk.py]: I had to comeback to my code to fix a problem in https://github.com/SUSE/BCI-tests/runs/7387709869?check_suite_focus=true

**Review by jlausuch — APPROVED** [review]

**Pull request merged**